code/modules/awaymissions/gateway.dm

```diff
@@ -1,252 +1,322 @@
+/datum/gateway_destination/proc/deactivate(obj/machinery/gateway/deactivated)
+	return
+
+/* Returns data used by gateway controller ui */
+/datum/gateway_destination/proc/get_ui_data()
+	. = list()
+	.["ref"] = REF(src)
+	.["name"] = name
+	.["availible"] = is_availible()
+	.["reason"] = get_availible_reason()
+	if(wait)
+		.["timeout"] = max(1 - (wait - world.time) / wait, 0)
+
+/* Destination is another gateway */
+/datum/gateway_destination/gateway
+	var/obj/machinery/gateway/G
+
+/* We set the target gateway target to activator gateway */
+/datum/gateway_destination/gateway/activate(obj/machinery/gateway/activated)
+	if(!G.target)
+		G.activate(activated)
+
+/* We turn off the target gateway if it's linked with us */
+/datum/gateway_destination/gateway/deactivate(obj/machinery/gateway/deactivated)
+	if(G.target == deactivated.destination)
+		G.deactivate()
+
+/datum/gateway_destination/gateway/is_availible()
+	return ..() && G.calibrated && !G.target && G.powered()
+
+/datum/gateway_destination/gateway/get_availible_reason()
+	. = ..()
+	if(!G.calibrated)
+		. = "Exit gateway malfunction. Manual recalibration required."
+	if(G.target)
+		. = "Exit gateway in use."
+	if(!G.powered())
+		. = "Exit gateway unpowered."
+
+/datum/gateway_destination/gateway/get_target_turf()
+	return get_step(G.portal,SOUTH)
+
+/datum/gateway_destination/gateway/post_transfer(atom/movable/AM)
+	. = ..()
+	AM.setDir(SOUTH)
+
+/* Special home destination, so we can check exile implants */
+/datum/gateway_destination/gateway/home
+
+/datum/gateway_destination/gateway/home/incoming_pass_check(atom/movable/AM)
+	if(isliving(AM))
+		if(check_exile_implant(AM))
+			return FALSE
+	else
+		for(var/mob/living/L in AM.contents)
+			if(check_exile_implant(L))
+				G.say("Rejecting [AM]: Exile implant detected in contained lifeform.")
+				return FALSE
+	if(AM.has_buckled_mobs())
+		for(var/mob/living/L in AM.buckled_mobs)
+			if(check_exile_implant(L))
+				G.say("Rejecting [AM]: Exile implant detected in close proximity lifeform.")
+				return FALSE
+	return TRUE
+
+/datum/gateway_destination/gateway/home/proc/check_exile_implant(mob/living/L)
+	for(var/obj/item/implant/exile/E in L.implants)//Checking that there is an exile implant
+		to_chat(L, "<span class='userdanger'>The station gate has detected your exile implant and is blocking your entry.</span>")
+		return TRUE
+	return FALSE
+
+
+/* Destination is one ore more turfs - created by landmarks */
+/datum/gateway_destination/point
+	var/list/target_turfs = list()
+	/// Used by away landmarks
+	var/id
+
+/datum/gateway_destination/point/get_target_turf()
+	return pick(target_turfs)
+
+/* Dense invisible object starting the teleportation. Created by gateways on activation. */
+/obj/effect/gateway_portal_bumper
+	var/obj/machinery/gateway/gateway
+	density = TRUE
+	invisibility = INVISIBILITY_ABSTRACT
+
+/obj/effect/gateway_portal_bumper/Bumped(atom/movable/AM)

```

Comment:
make it optionally directional where it can only be entered from a specific direction?